modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,184 @@
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.translation.translator.Translator;
+import com.liferay.translation.translator.TranslatorPacket;
+import com.liferay.translation.translator.deepl.internal.configuration.DeepLTranslatorConfiguration;
+import com.liferay.translation.translator.deepl.internal.constants.DeepLConstants;
+import com.liferay.translation.translator.deepl.internal.model.SupportedLanguage;
+import com.liferay.translation.translator.deepl.internal.model.TranslateResponse;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * @author Yasuyuki Takeo
+ */
+@Component(
+    configurationPid = "com.liferay.translation.translator.deepl.internal.configuration.DeepLTranslatorConfiguration",
+    service = Translator.class
+)
+public class DeepLTranslator implements Translator {
+  @Override
+  public boolean isEnabled(long companyId) throws ConfigurationException {
+    DeepLTranslatorConfiguration
+        deepLTranslatorConfiguration =
+        _configurationProvider.getCompanyConfiguration(
+            DeepLTranslatorConfiguration.class, companyId);
+
+    return deepLTranslatorConfiguration.enabled() &&
+        !Validator.isBlank(deepLTranslatorConfiguration.authKey()) &&
+        !Validator.isBlank(deepLTranslatorConfiguration.url());
+  }
+
+  private String _getLanguageCode(String languageId) {
+    List<String> list = StringUtil.split(languageId, CharPool.UNDERLINE);
+
+    return list.get(0);
+  }
+
+  protected Boolean _verifyLanguage(List<String> deepLLanguages, List<String> compareLanguages) {
+    for (String lang : compareLanguages) {
+      if (Collections.disjoint(deepLLanguages, Arrays.asList(lang))) {
+        _log.error("DeepL does not support " + lang + ". Abort processing translation.");
+        return false;
+      }
+    }
+    return true;
+  }
+
+  protected List<String> _getSupportedLanguages(
+      DeepLTranslatorConfiguration deepLTranslatorConfiguration) {
+    List<SupportedLanguage> supportedLanguages
+        = null;
+    try {
+      supportedLanguages = _deepLClient.verifySupportedLanguage(
+          deepLTranslatorConfiguration.authKey(),
+          DeepLConstants.TARGET, DeepLConstants.SUPPORTED_LANGUAGE_INQ_URL
+      );
+    } catch (IOException e) {
+      _log.error("Failed to call supported language list." + System.lineSeparator() + e.getLocalizedMessage());
+      return new ArrayList<>();
+    }
+
+    List<String> languages = new ArrayList<>();
+    supportedLanguages.forEach(sl -> {
+      languages.add(sl.language);
+    });
+    return languages;
+  }
+
```

Comment:
@hhuijser Auto sf is green, but the format of this pr is all wrong. Shouldn't have it failed?